src/torch_models.py

```diff
@@ -146,6 +147,107 @@ def _fit(self, inputs: Array, outputs: Array) -> None:
         logging.info(f"Loaded best model with loss: {loss:.5f}")
 
 
+class ImplicitMLPRegressor(Regressor):
+    """A regressor implemented via an energy function (binary classifier).
```

Comment:
let's fix this comment to apologize, maybe saying like, currently we use a binary classifier, but this is not ideal